### ISSUE TITLE
flutter_firebase_chat_core function get list of users through FirebaseChatCore.instance.users(); is not returning list of users in android OS version 11 and 10 

### ISSUE THREAD

**Issue by DikshaCT27** [flyerhq/flutter_firebase_chat_core#78]
# General

## What bug do you experience? 🐞
Provide a clear and concise description of what the bug is.

## How can it be reproduced? 🤔
A few steps to define where does the bug occur.
Step 1. ...
Step 2. ... etc.

## What behavior is expected? 💡

<br />

# Extras

## Screenshots or videos 📸
If applicable, add screenshots or videos to help explain your problem.

## Code snippets 📝
If applicable, add code samples to help explain your problem.

```dart
// Your snippet here...
```

<br />

# Environment info

Please specify the flutter, flutter_firebase_chat_core versions.

flutter: ...
flutter_firebase_chat_core: ...

`flutter doctor -v` output 👇

```sh
```

<br />

# Platform

Device (e.g. Android emulator, iOS simulator, Samsung Galaxy S21): ...

OS version (e.g. iOS 14.5, Android 11): ...

<br />

# Additional context

<!-- Thank you for making us better 🤝--!>

**Comment by DikshaCT27** [comment]
var userToRoom = FirebaseChatCore.instance.users();
userToRoom.listen((listOfStrings) async {
              // From this point you can use listOfStrings as List<String> object
              // and do all other business logic you want

              for (var con in listOfStrings) {
                // print(str);
                if (con.metadata!
                    .containsValue(widget.patronData['email'])) {
                  sendUserToRoom = myString;
                  print(sendUserToRoom);
                  break;
                } else {}
              }
}

Here userToRoom has 6 entries but retrieves an Empty stream when trying in OS versions 11 and 10.

**Comment by DikshaCT27** [comment]
An instance of '_EmptyStream<List<User>>' is what I am getting as a print statement in the userToRoom variable.

**Comment by demchenkoalex** [comment]
Maybe old version? Can you try on new, Flutter 3 version?

**Comment by DikshaCT27** [comment]
@demchenkoalex I am trying this on Flutter 3.0.1 already, but getting issue.

**Comment by demchenkoalex** [comment]
Does it work on iOS? Or Android versions 9 or 12?

**Comment by DikshaCT27** [comment]
No , checked in android version 10,11,12 not working

**Comment by demchenkoalex** [comment]
And iOS? If iOS does not work as well I think it might be something on your side. I do receive users in the example app.

**Comment by demchenkoalex** [comment]
I don't have the data you have, and can't use your code. Your path should be - checking that simple `FirebaseChatCore.instance.users` works (similarly to the example) and then debugging it from that point.

**Comment by DikshaCT27** [comment]
FirebaseChatCore.instance.users is returning An instance of '_EmptyStream<List>' this and hence I am not able to create a room for the user .

**Comment by demchenkoalex** [comment]
Okay, but in the example it returns a list of users normally. Then it is a matter of mapping users from your Firebase store, something is off there, probably users can't be mapped, that means some values are off? I think I had this even if at least one entity is broken, the whole list will be empty. How many users do you have and are you able to verify they can be mapped correctly?

**Comment by DikshaCT27** [comment]
@demchenkoalex I figured out this issue and bingo, I found the bug. Thanks for elaborating. Resolved.

**Comment by DikshaCT27** [comment]
@demchenkoalex It's not working again , I tried getting users list from firebase method its returning and then created a room through flyer chat method createRoom it's saying "User doesn't exist"

`var userToRoom =
                FirebaseFirestore.instance.collection('users').snapshots();
            print(userToRoom);
            userToRoom.listen((listOfStrings) async {
              for (var myString in listOfStrings.docs) {
                // print(str);
                var email = myString.get('metadata');
                if (email['email'] == widget.patronData['email']) {
                  sendUserToRoom = myString;
                  print(sendUserToRoom);
                  break;
                } else {}
              }
              Map str = sendUserToRoom.data();
              var user = types.User(
                      id: sendUserToRoom.id,
                      firstName: str['firstName'],
                      lastName: str['lastName'],
                      imageUrl: str['imageUrl'])
                  .toJson();
              print(user);
              final room = await FirebaseChatCore.instance.createRoom(
                  types.User(
                      id: sendUserToRoom.id,
                      firstName: str['firstName'],
                      lastName: str['lastName'],
                      imageUrl: str['imageUrl']));`

**Comment by DikshaCT27** [comment]
It's sudden sometimes it's returning users and sometimes not

**Comment by demchenkoalex** [comment]
for the future, please use a code block with a formatting, cause it is hard to follow that piece of code above. You can do this by typing ```dart, then on the next line your code, then close with 3 backticks again. (it will look like the code below I pasted)

There is one thing I realised regarding the `createRoom` function - it actually uses only ID inside, so you could simplify code by removing all other fields, like firstName, lastName, imageUrl from the call to `createRoom` - e.g.

```dart
final room = await FirebaseChatCore.instance.createRoom(types.User(id: sendUserToRoom.id))
```

Next, I not sure I understand all of those strings transformations you have, when you listen to the `FirebaseChatCore.instance.users()` snapshots, in the data you have `User` types - I see you need to check `metadata` to select a particular user, why not just do

```dart
user.metadata?['email'] == widget.patronData['email']
```

and then you have a `User` without any modifications, faster and easier.

Next, you can always look at the source code (it is super easy), for example [users stream](https://github.com/flyerhq/flutter_firebase_chat_core/blob/main/lib/src/firebase_chat_core.dart#L422) - see, there is no dark magic inside, so in case you sometimes don't get somethings, please check your transformations and/or what you have in Firebase (is it confirming to the `User` type) - all should be correctly working.

**Comment by DikshaCT27** [comment]
@demchenkoalex Thank you very much for the formatting tip, I am new with it.

Regarding your suggestion not to use firstName, LastName let me try this out. Thanks for the clarification.
